src/test/java/io/managed/services/test/kafka/KafkaMgmtAPITest.java

```diff
@@ -218,6 +218,123 @@ public void testCreateTopics() {
         kafkaInstanceApi.createTopic(metricTopicPayload);
     }
 
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        log.info("Max partition limit per given instance: {}", maxPartitionLimit);
+
+        log.info("Attempt to create topic '{}', with too many partitions: {}", topicName, maxPartitionLimit + 1);
+        var payload = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit + 1));
+        assertThrows(ApiGenericException.class,
+                () -> kafkaInstanceApi.createTopic(payload));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedParitionLimitByTopicConfiguration() {
+
+        var topicName = "topic-part-exceed-conf";
+        KafkaInstanceApiUtils.applyTopic(kafkaInstanceApi, topicName);
+
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        assertThrows(ApiGenericException.class,
+                () -> KafkaInstanceApiUtils.updateTopicPartition(kafkaInstanceApi, topicName, maxPartitionLimit + 1));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testTotalPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-total-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        final int currentPartitionCount = KafkaInstanceApiUtils.getPartitionCountTotal(kafkaInstanceApi);
+
+        var payloadTopicMax = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit));
+
+        // I can exceed the max limit with single new topic
+        if (currentPartitionCount > 0) {
+            log.info("Trying to create topic with {} partitions, while total partition limit is {}, and instance already has {} partitions", maxPartitionLimit, maxPartitionLimit, currentPartitionCount);
+            assertThrows(ApiGenericException.class, () -> kafkaInstanceApi.createTopic(payloadTopicMax));
+        } else {
+            // create two topics, first will reach the max limit, creation of second will exceed it.
```

Comment:
I'd suggest https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/Assumptions.html assuming that maxPartitionLimit - currentPartitionCount > 0 (partitions available precondition)

Then divide (maxPartitionLimit - currentPartitionCount) by 2 and have the first topic use halve, and the second topic try to create halve+1.